crates/integrations/datafusion/Cargo.toml

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "iceberg-integrations-datafusion"
+version = { workspace = true }
+edition = { workspace = true }
+homepage = { workspace = true }
+rust-version = { workspace = true }
+
+categories = ["database"]
+description = "Apache Iceberg Integrations Datafusion"
+repository = { workspace = true }
+license = { workspace = true }
+keywords = ["iceberg", "integrations", "datfusion"]
+
+[features]
+default = []
+datafusion = ["dep:datafusion"]
```

Comment:
Sorry I'm a little confusing here, as a crate for integration with datafusion, why this is a feature rather enabled by default?

Reply:
Here I'm probably even more confused; I tried this based on the review by @tshauck - but I have no experience with this and I'm not sure how to structure the crate, in order to allow a possible end-user to download only the integration dependency he needs e.g. iceberg-datafusion or iceberg-polars. etc.

So here I'm hoping for feedback and help.

Reply:
I believe we should have `integrations/datafusion` and `integrations/polars`. So this crate `iceberg-integrations-datafusion` should enable `datafusion` by default, and doesn't need switch.

Reply:
Sorry for muddying the waters. I tried to suggest not going the feature route based [here](https://github.com/apache/iceberg-rust/pull/324#discussion_r1562767139) (near the end). I was a little confused by this project having all dependencies in the workspace's Cargo.toml even if only used by one crate.

Reply:
> I believe we should have `integrations/datafusion` and `integrations/polars`. So this crate `iceberg-integrations-datafusion` should enable `datafusion` by default, and doesn't need switch.

@Xuanwo 
thank you - then I guess my last commit fixed it.